### PR TITLE
Add support for sonarjs

### PR DESCRIPTION
**Pull request by Julien-R44**
https://github.com/SonarSource/eslint-plugin-sonarjs

- Generated rules
- Added extensions/configurations
- Removed useless documentation

Related to #131

### CODE REVIEW

File: src/extends/index.d.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import type { VueExtensions } from './eslint-plugin-vue';
 import type { VuePugExtensions } from './eslint-plugin-vue-pug';
 import type { IntlifyVueI18nExtensions } from './intlify-vue-i18n';
 import type { TypescriptEslintExtensions } from './typescript-eslint';
+import type { SonarjsExtensions } from './eslint-plugin-sonarjs';
```
Comment (Shinigami92) [inline src/extends/index.d.ts:13]: I think you can fix things like these with VSCode Organize Imports

Comment (Julien-R44) [reply src/extends/index.d.ts]: Okay actually it's via a prettier plugin that you sort the imports. But in my case it doesn't seem to want to use it. I'll try to investigate

**Comment by xiaoxiangmoe** [comment]
@Julien-R44 There is also another related eslint plugin https://github.com/un-ts/eslint-plugin-sonar

**Comment by Julien-R44** [comment]
Thanks for sharing, I didn't know. Would also be cool to have

File: package.json [hunk 2]
```diff
@@ -55,6 +55,7 @@
     "@types/node": "~18.7.23",
     "@types/prettier": "~2.7.1",
     "@typescript-eslint/eslint-plugin": "~5.38.1",
+    "@typescript-eslint/experimental-utils": "^5.38.1",
```
Comment (Shinigami92) [inline package.json:58]: Why is this needed?

Comment (Julien-R44) [reply package.json:58]: I can't remember why ahah. I'm sorry for the delay @Shinigami92 I have a lot of stuff to do right now 🥲

I think I'll get back to the project in a week or two and resume theses PR

However, what I wanted to do was to finish our patch module, and once merged, regenerate the types for the plugins I proposed PRs for
